perun-core/src/main/java/cz/metacentrum/perun/core/entry/ResourcesManagerEntry.java

```diff
@@ -542,6 +542,23 @@ public void assignService(PerunSession sess, Resource resource, Service service)
 		getResourcesManagerBl().assignService(sess, resource, service);
 	}
 
+	@Override
+	public void assignServices(PerunSession sess, Resource resource, List<Service> services) throws PrivilegeException, ResourceNotExistsException, ServiceNotExistsException, ServiceAlreadyAssignedException, WrongAttributeValueException, WrongReferenceAttributeValueException {
+		Utils.checkPerunSession(sess);
+		getResourcesManagerBl().checkResourceExists(sess, resource);
+
+		// Authorization
+		if (!AuthzResolver.isAuthorized(sess, Role.FACILITYADMIN, resource)) {
+			throw new PrivilegeException(sess, "assignServices");
+		}
+
+		for (Service service : services) {
```

Comment:
It would be nice to have these checks before the authorization check. The new authorization model needs to be sure that all objects exist when the authorization checks are performed.

Reply:
Moved.